server/src/projects/project-applicants/project-applicant.controller.ts

```diff
@@ -0,0 +1,85 @@
+import { Controller, Patch, Body, Param, Post, UseInterceptors, UseGuards, UsePipes, Delete, HttpException, HttpStatus } from '@nestjs/common';
+import { pick } from 'underscore';
+import { CrmService } from '../../crm/crm.service';
+import { JsonApiSerializeInterceptor } from '../../json-api-serialize.interceptor';
+import { AuthenticateGuard } from '../../authenticate.guard';
+import { JsonApiDeserializePipe } from '../../json-api-deserialize.pipe';
+import { PROJECTAPPLICANT_ATTRS } from './project-applicants.attrs';
+import { CONTACT_ATTRS } from '../../contact/contacts.attrs';
+
+const ACTIVE_CODE = 1;
+
+@UseInterceptors(new JsonApiSerializeInterceptor('project-applicants', {
+  id: 'dcp_projectapplicantid',
+  attributes: [
+    ...PROJECTAPPLICANT_ATTRS,
+
+    'contact',
+  ],
+
+  contact: {
+    ref: 'contactid',
+    attributes: CONTACT_ATTRS,
+  },
+}))
+@UseGuards(AuthenticateGuard)
+@UsePipes(JsonApiDeserializePipe)
+@Controller('project-applicants')
+export class ProjectApplicantController {
+  constructor(private readonly crmService: CrmService) {}
+
+  @Post('/')
+  async create(@Body() body) {
+    const allowedAttrs = pick(body, PROJECTAPPLICANT_ATTRS);
+    const email = body.emailaddress;
+    const projectId = body.project;
+    const fullName = body.dcp_name;
+
+    let contactId;
+
+    // check if contact already exists for this email address
+    const { records } = await this.crmService.get('contacts', `
+      $select=${CONTACT_ATTRS.join(',')}
+      &$filter=startswith(emailaddress1, '${email}')
+        and statuscode eq ${ACTIVE_CODE}
+      &$top=1
+    `);
+
+    let newContact = {
+      contactid: null,
+    };
+
+    if (records.length > 0) {
+      contactId = records[0].contactid;
+    } else {
+      const regex = /^([\w\-]+)/g;
+      const firstName = fullName.match(regex)[0];
+      const lastName = fullName.replace(`${firstName} `, '');
+      newContact = await this.crmService.create(`contacts`, {
+        'emailaddress1': email,
+        'firstname': firstName,
+        'lastname': lastName,
+      });
+      contactId = newContact.contactid;
+    }
+
+    if (!body.emailaddress || !body.project) throw new HttpException(
+      'Missing email address or project id',
```

Comment:
Can you add `title`, `code` and `detail` properties here? 
like 
```
{
  code: 'ADD_APPLICANT_ERROR'
  title: 'Add Applicant failed',
  detail: 'Cannot add applicant because missing ....',
}
```